pkg/overlay/cache.go

```diff
@@ -130,3 +131,16 @@ func (cache *Cache) Put(ctx context.Context, nodeID storj.NodeID, value pb.Node)
 
 	return cache.db.Put(nodeID.Bytes(), data)
 }
+
+// ConnFailure implements the Transport Observer `ConnFailure` function
+func (cache *Cache) ConnFailure(ctx context.Context, node *pb.Node, err error) {
+	// noop until we figure out what to do on ConnFailure
+}
+
+// ConnSuccess implements the Transport Observer `ConnSuccess` function
+func (cache *Cache) ConnSuccess(ctx context.Context, node *pb.Node) {
+	err := cache.Put(ctx, node.Id, *node)
+	if err != nil {
+		zap.L().Debug("error putting node o cache:", zap.Error(err))
```

Comment:
I should say "add logger to struct, call here" but this is an interesting case.  You might also want to log some extra information about the context here, as knowing the class that created the error doesn't tell you a lot about where the error came from.

Reply:
after `cache.Put`, could you add the following?
```
_, err := cache.statdb.UpdateUptime(ctx, node.Id, true)
if err != nil {
	// log statdb error
}
```